### ISSUE TITLE
Build fail

### ISSUE THREAD

**Issue by brunosmartin** [kadoshms/ionic2-autocomplete#20]
This is the easy way to reproduce, but when I run ionic build android --build --release I have the same error.

```
# npm run build

> ionic2-auto-complete@1.4.1-rc2 build /home/xxxxx/devel/ionic2-autocomplete
> ngc -p tsconfig.json

Unexpected value 'AutoCompleteItemComponent in /home/xxxxx/devel/ionic2-autocomplete/src/autocompleteitem.component.ts' declared by the module 'AutoCompleteModule in /home/xxxx/devel/ionic2-autocomplete/index.ts'. Please add a @Pipe/@Directive/@Component annotation.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! ionic2-auto-complete@1.4.1-rc2 build: `ngc -p tsconfig.json`
npm ERR! Exit status 1
```

**Comment by kadoshms** [comment]
Could you please attach your `ionic info` ?

**Comment by callym** [comment]
I get the same error using 1.4.2-rc2

```
ionic cordova build android --release --prod
Running app-scripts build: --prod --iscordovaserve --externalIpRequired --nobrowser

[11:39:09]  build prod started ...
[11:39:09]  clean started ...
[11:39:09]  clean finished in 4 ms
[11:39:09]  copy started ...
[11:39:09]  ngc started ...
{ Error: Unexpected value 'AutoCompleteItemComponent in D:/programming/myapp/node_modules/ionic2-auto-complete/src/autocompleteitem.component.ts' declared by the module 'AutoCompleteModule in D:/programming/myapp/node_modules/ionic2-auto-complete/index.ts'. Please add a @Pipe/@Directive/@Component annotation.
    at Error (native)
    at syntaxError (D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:1513:34)
    at D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:14028:40
    at Array.forEach (native)
    at CompileMetadataResolver.getNgModuleMetadata (D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:14010:54)
    at addNgModule (D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:22555:58)
    at D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:22566:14
    at Array.forEach (native)
    at _createNgModules (D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:22565:26)
    at analyzeNgModules (D:\programming\myapp\node_modules\@angular\compiler\bundles\compiler.umd.js:22440:14) hasBeenLogged: undefined, isFatal: undefined, name: 'Error' }
```

```
ionic info

global packages:

    @ionic/cli-utils : 1.0.0-rc.1
 You have been opted out of telemetry. To change this, run: cordova telemetry on. 6.5.0
    Ionic CLI        : 3.0.0-rc.1

local packages:

    @ionic/app-scripts              : 1.3.6
    @ionic/cli-plugin-cordova       : 1.0.0-rc.1
    @ionic/cli-plugin-ionic-angular : 1.0.0-rc.1
    Ionic Framework                 : ionic-angular 3.1.1

System:

    Node       : v6.5.0
    OS         : Windows 10
    Xcode      : not installed
    ios-deploy : not installed
    ios-sim    : not installed
```

**Comment by KyleMcNutt** [comment]
I'm also seeing this same error:

```
Ionic Framework Version: 2.3.0
Ionic CLI Version: 2.2.3
Ionic App Lib Version: 2.2.1
Ionic App Scripts Version: 1.1.4
ios-deploy version: 1.9.1 
ios-sim version: 5.0.13 
OS: OS X El Capitan
Node Version: v7.8.0
Xcode version: Xcode 8.2.1 Build version 8C1002
```

**Comment by KyleMcNutt** [comment]
From what I can tell, an `AOT` build doesn't like the use of the `AutoCompleteItem` decorator in order to set the template. I removed that decorator and just had it be a plain `Component` with just the default template (which is all I needed in my case) and this error went away.

**Comment by kadoshms** [comment]
@KyleMcNutt your'e probably right. 
I am planning to remove this feature (and use the plain old Component) in the next release (in this weekend).
This is a remnant from the days before AOT....

**Comment by KyleMcNutt** [comment]
@kadoshms Sounds good, looking forward to the next release.

**Comment by kadoshms** [comment]
Fixed in abb818bc5d20491fdfc2475c4c28d1abd8b76478 and published as version 1.4.3-beta.

**Comment by KyleMcNutt** [comment]
@kadoshms This worked great! Thanks for the quick update!
